src/App.tsx

```diff
@@ -15,71 +16,108 @@ export const goods = [
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  clearSelectedGood = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  selectGood(good: string) {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    if (!selectedGood.length) {
+      this.setState({ selectedGood: good });
+    } else if (selectedGood !== good) {
+      this.setState({ selectedGood: good });
+    } else {
+      this.setState({ selectedGood: '' });
+    }
+  }
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1
+          className={
+            classNames(
+              'title',
+              { 'is-flex is-align-items-center': selectedGood.length > 0 },
+            )
+          }
+        >
+          {
+            selectedGood.length > 0
+              ? `${selectedGood} is selected`
+              : 'No goods selected'
+          }
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+          {selectedGood.length > 0
+            && (
+            /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.clearSelectedGood}
+              />
+            )}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+        <table className="table">
+          <tbody>
+            {goods.map((good) => (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={classNames(
+                  '',
+                  {
+                    'has-background-success-light': selectedGood === good,
```

Comment:
let's save result of this check to constant, you've used it 4 times, no need to repeat yourself